extensions/wikidata/tests/src/org/openrefine/wikidata/qa/MockConstraintFetcher.java

```diff
@@ -211,10 +218,42 @@ public boolean hasDiffWithinRange(PropertyIdValue pid) {
         return true;
     }
 
-    public Map<PropertyIdValue, List<Value>> getParamConflictsWith(PropertyIdValue pid) {
-        Map<PropertyIdValue, List<Value>> propertyIdValueListMap = new HashMap<>();
-        List<Value> items = Arrays.asList(conflictingStatementValue, null);
-        propertyIdValueListMap.put(conflictingStatementPid, items);
-        return propertyIdValueListMap;
+    @Override
+    public Stream<Statement> getConstraintsByType(PropertyIdValue pid, String qid) {
+        EntityIdValue entityIdValue = Datamodel.makeWikidataItemIdValue("Q21502838");
+        PropertyIdValue propertyIdValue = Datamodel.makeWikidataPropertyIdValue("P2302");
+        Snak snak = Datamodel.makeValueSnak(propertyIdValue,entityIdValue);
+
+        PropertyIdValue property = Datamodel.makeWikidataPropertyIdValue("P2306");
+        Value propertyValue = Datamodel.makeWikidataPropertyIdValue("P31");
+        Snak snak1 = Datamodel.makeValueSnak(property,propertyValue);
+        List<Snak> group1 = new ArrayList<>();
+        group1.add(snak1);
+
+        PropertyIdValue item = Datamodel.makeWikidataPropertyIdValue("P2305");
+        Value itemValue = Datamodel.makeWikidataItemIdValue("Q5");
+        Snak snak2 = Datamodel.makeValueSnak(item,itemValue);
+        List<Snak> group2 = new ArrayList<>();
+        group2.add(snak2);
+
+        SnakGroup snakGroup1 = Datamodel.makeSnakGroup(group1);
+        SnakGroup snakGroup2 = Datamodel.makeSnakGroup(group2);
+
+        List<SnakGroup> list = new ArrayList<>();
+        list.add(snakGroup1);
+        list.add(snakGroup2);
+        Claim claim = Datamodel.makeClaim(entityIdValue,snak, list);
+
+        Reference reference = Datamodel.makeReference(list);
+        List<Reference> referenceList = new ArrayList<>();
+        referenceList.add(reference);
+
+        Statement statement = Datamodel.makeStatement(claim,referenceList, StatementRank.NORMAL, "P50$77BD7FE4-C051-4776-855C-543F0CE697D0");
```

Comment:
It might be worth ensuring that the statement id uses the same property id as the claim ("P2302" instead of "P50"), because there are plans to validate statement ids in forthcoming versions of WDTK.